sql/moz-fx-data-shared-prod/adjust_derived/adjust_derived_v1/query.py

```diff
@@ -0,0 +1,305 @@
+from argparse import ArgumentParser
+
+import requests
+from google.cloud import bigquery
+
+API_URI = "api.adjust.com"
+ENDPOINT = "kpis"
+API_VERSION = "v1"
+
+"""This is a list of dictionaries.
+Keys are the app_name, and app_token in the form:
+{"app_name":"<specific app name from Adjust dash>". "app_token":"<unique Adjust app token>}"
+Mozilla has 16 apps on Adjust.
+"""
+
+APP_TOKEN_LIST = os.environ.get("ADJUST_APP_TOKEN_LIST")
+
+"""This is Marlene's personal Adjust API Token. We don't have a special service account API Token."""
+API_TOKEN = os.environ.get("ADJUST_API_TOKEN")
+
+CSV_FIELDS = []
+CSV_FIELDS = [
+    "date",
+    "app",
+    "app_token",
+    "network",
+    "network_token",
+    "campaign",
+    "campaign_token",
+    "adgroup",
+    "adgroup_token",
+    "creative",
+    "creative_token",
+    "country",
+    "os",
+    "device",
+    "clicks",
+    "installs",
+    "limit_ad_tracking_install_rate",
+    "click_conversion_rate",
+    "impression_conversion_rate",
+    "sessions",
+    "daus",
+    "waus",
+    "maus",
+]
+
+
+def read_json(filename: str) -> dict:
+    """Read JSON file."""
+    with open(filename, "r") as f:
+        data = json.loads(f.read())
+    return data
+
+
+def write_dict_to_csv(json_data, filename):
+    """Write a dictionary to a csv."""
+    with open(filename, "w") as out_file:
+        dict_writer = csv.DictWriter(out_file, CSV_FIELDS)
+        dict_writer.writeheader()
+        dict_writer.writerows(json_data)
+
+
+def download_adjust_kpi_data(date, api_token, app_token):
+    """Download data from Adjust - API token and APP tokens are called here."""
+    start_date = date
+    end_date = date
+
+    kpis = [
+        "clicks",
+        "installs",
+        "limit_ad_tracking_install_rate",
+        "click_conversion_rate",
+        "impression_conversion_rate",
+        "sessions",
+        "daus",
+        "waus",
+        "maus",
+    ]
+    groupings = [
+        "days",
+        "apps",
+        "networks",
+        "campaigns",
+        "adgroups",
+        "creatives",
+        "countries",
+        "os_names",
+        "device_types",
+    ]
+    # getting overview metrics for different kpis / Deliverables
+    url = f"https://api.adjust.com/kpis/v1/{app_token}"  # overview
+    url_params = f"start_date={start_date}&end_date={end_date}&kpis={','.join(kpis)}&grouping={','.join(groupings)}"
+    headers = {
+        "Authorization": f"Token token={api_token}",
+    }
+
+    response = requests.get(url, headers=headers, params=url_params)
+
+    return response
+
+
+def clean_json(adjust_response_text, adjust_list_part):
+    """JSON sometimes has missing keys, need to clean up the data."""
+    fields_list = []
+    with tempfile.NamedTemporaryFile() as tmp_json:
+        with open(tmp_json.name, "w") as f_json:
+            f_json.write(adjust_response_text)
+
+            query_export = read_json(f_json.name)
+
+            for date in query_export["result_set"]["dates"]:
```

Comment:
You could separate the code from here below into another function e.g. get_query_fields() that would run for each date. Mainly for readability and to separate what each part of the code does. Optionally you can do that in a later PR, your call!

Reply:
In this function you can use the [recommended practice for unexisting keys](https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_get_to_return_a_default_value_from_a_dictionary.html#best-practice).

Reply:
Thank you for this! Will update code